init/pkg/utils.go

```diff
@@ -0,0 +1,271 @@
+		hbPortParam         int32
+		fabricTLSPortParam  int32
+		fabricPortParam     int32
+	)
+
+	asConfig := aeroCluster.Spec.AerospikeConfig
+
+	if _, serviceTLSPort := asdbv1beta1.GetServiceTLSNameAndPort(asConfig); serviceTLSPort != nil {
+		serviceTLSPortParam = int32(*serviceTLSPort)
+	}
+
+	if servicePort := asdbv1beta1.GetServicePort(asConfig); servicePort != nil {
+		servicePortParam = int32(*servicePort)
+	}
+
+	if _, hbTLSPort := asdbv1beta1.GetHeartbeatTLSNameAndPort(asConfig); hbTLSPort != nil {
+		hbTLSPortParam = int32(*hbTLSPort)
+	}
+
+	if hbPort := asdbv1beta1.GetHeartbeatPort(asConfig); hbPort != nil {
+		hbPortParam = int32(*hbPort)
+	}
+
+	if _, fabricTLSPort := asdbv1beta1.GetFabricTLSNameAndPort(asConfig); fabricTLSPort != nil {
+		fabricTLSPortParam = int32(*fabricTLSPort)
+	}
+
+	if fabricPort := asdbv1beta1.GetFabricPort(asConfig); fabricPort != nil {
+		fabricPortParam = int32(*fabricPort)
+	}
+
+	networkInfo := &networkInfo{
+		multiPodPerHost:  aeroCluster.Spec.PodSpec.MultiPodPerHost,
+		networkPolicy:    aeroCluster.Spec.AerospikeNetworkPolicy,
+		podPort:          servicePortParam,
+		podTLSPort:       serviceTLSPortParam,
+		heartBeatPort:    hbPortParam,
+		heartBeatTLSPort: hbTLSPortParam,
+		fabricPort:       fabricPortParam,
+		fabricTLSPort:    fabricTLSPortParam,
+		hostNetwork:      aeroCluster.Spec.PodSpec.HostNetwork,
+		hostIP:           os.Getenv("MY_HOST_IP"),
+		podIP:            os.Getenv("MY_POD_IP"),
+	}
+
+	if err := setHostPortEnv(k8sClient, podName, aeroCluster.Namespace, networkInfo); err != nil {
+		return nil, err
+	}
+
+	return networkInfo, nil
+}
+
+func GetNodeIDFromPodName(podName string) (nodeID string, err error) {
+	parts := strings.Split(podName, "-")
+	if len(parts) < 3 {
+		return "", fmt.Errorf("failed to get nodeID from podName %s", podName)
+	}
+	// Podname format stsname-ordinal
+	// stsname ==> clustername-rackid
+	nodeID = parts[len(parts)-2] + "a" + parts[len(parts)-1]
+
+	return nodeID, nil
+}
+
+func getRack(podName string, aeroCluster *asdbv1beta1.AerospikeCluster) (*asdbv1beta1.Rack, error) {
+	res := strings.Split(podName, "-")
+
+	rackID, err := strconv.Atoi(res[len(res)-2])
+	if err != nil {
+		return nil, err
+	}
+
+	logrus.Info("Checking for rack in rackConfig ", "rack-id=", rackID)
+
+	racks := aeroCluster.Spec.RackConfig.Racks
+	for idx := range racks {
+		rack := &racks[idx]
+		if rack.ID == rackID {
+			return rack, nil
+		}
+	}
+
+	return nil, fmt.Errorf("rack with rack-id %d not found", rackID)
+}
+
+func (initp *InitParams) makeWorkDir() error {
+	if initp.workDir != "" {
+		defaultWorkDir := filepath.Join("workdir", "filesystem-volumes", initp.workDir)
+
+		requiredDirs := [3]string{"smd", "usr/udf/lua", "xdr"}
+		for _, d := range requiredDirs {
+			toCreate := filepath.Join(defaultWorkDir, d)
+			logrus.Info("Creating directory ", toCreate)
+
+			if err := os.MkdirAll(toCreate, 0644); err != nil { //nolint:gocritic // file permission
```

Comment:
I think we should add execute permission too for directories.
0755.
Check other places as well.